de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java

```diff
@@ -67,22 +77,53 @@ public void applyJoinRelation(FskPortObject fskObj, List<JoinRelationAdvanced> j
             String resourcePath = sourceModel.getGeneratedResourcesDirectory().get()

```

Comment:
getGeneratedResourcesDirectory() returns an Optional that may be empty and needs to be checked.